### DISCUSSION TITLE
Dynamically generated pages (e.g. one page per db record)

### DISCUSSION THREAD

**Discussion by honzajavorek** [mkdocs/mkdocs#3118] [Q&A] [answered]
What is the best practice to have some pages generated dynamically, e.g. from database? I already use plugins and custom hooks to enrich my static markdown pages with some data, but now the task is to be able to have a completely dynamic list of pages.

My first, naive thought would be to have a custom script, basically unrelated to the mkdocs instance, which reads from database and generates the `.md` files as if I wrote them myself. That might cause some trouble with keeping the navigation up to date, but otherwise I guess it could work.

Any other ideas? Any existing plugins? Am I overlooking something?

**Comment by pawamoy** [comment, marked as answer]
I believe the [mkdocs-gen-files](https://github.com/oprypin/mkdocs-gen-files) plugin achieves what you need.

**Reply by honzajavorek** [reply]
That looks like something that could help me. Thanks!
